pkg/test/selectorsyncset/selectorsyncset.go

```diff
@@ -0,0 +1,96 @@
+package selectoryncset
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/test/generic"
+)
+
+// Option defines a function signature for any function that wants to be passed into Build
+type Option func(*hivev1.SelectorSyncSet)
+
+// Build runs each of the functions passed in to generate the object.
+func Build(opts ...Option) *hivev1.SelectorSyncSet {
+	retval := &hivev1.SelectorSyncSet{}
+	for _, o := range opts {
+		o(retval)
+	}
+
+	return retval
+}
+
+type Builder interface {
+	Build(opts ...Option) *hivev1.SelectorSyncSet
+
+	Options(opts ...Option) Builder
+
+	GenericOptions(opts ...generic.Option) Builder
+}
+
+func BasicBuilder() Builder {
+	return &builder{}
+}
+
+func FullBuilder(name string, typer runtime.ObjectTyper) Builder {
+	b := &builder{}
+	return b.GenericOptions(
+		generic.WithTypeMeta(typer),
+		generic.WithResourceVersion("1"),
+		generic.WithName(name),
+	)
+}
+
+type builder struct {
+	options []Option
+}
+
+func (b *builder) Build(opts ...Option) *hivev1.SelectorSyncSet {
+	return Build(append(b.options, opts...)...)
+}
+
+func (b *builder) Options(opts ...Option) Builder {
+	return &builder{
+		options: append(b.options, opts...),
+	}
+}
+
+func (b *builder) GenericOptions(opts ...generic.Option) Builder {
+	options := make([]Option, len(opts))
+	for i, o := range opts {
+		options[i] = Generic(o)
+	}
+	return b.Options(options...)
+}
+
+// Generic allows common functions applicable to all objects to be used as Options to Build
+func Generic(opt generic.Option) Option {
+	return func(selectorSyncSet *hivev1.SelectorSyncSet) {
+		opt(selectorSyncSet)
+	}
+}
+
+// WithName sets the object.Name field when building an object with Build.
+func WithName(name string) Option {
+	return Generic(generic.WithName(name))
+}
+
+// WithNamespace sets the object.Namespace field when building an object with Build.
+func WithNamespace(namespace string) Option {
+	return Generic(generic.WithNamespace(namespace))
+}
+
+func WithLabelSelector(labelKey, labelValue string) Option {
+	return func(selectorSyncSet *hivev1.SelectorSyncSet) {
+		if selectorSyncSet.Spec.ClusterDeploymentSelector.MatchLabels == nil {
+			selectorSyncSet.Spec.ClusterDeploymentSelector.MatchLabels = make(map[string]string, 1)
+		}
+		selectorSyncSet.Spec.ClusterDeploymentSelector.MatchLabels[labelKey] = labelValue
+	}
+}
+
+func WithApplyMode(applyMode hivev1.SyncSetResourceApplyMode) Option {
+	return func(selectorSyncSet *hivev1.SelectorSyncSet) {
+		selectorSyncSet.Spec.ResourceApplyMode = applyMode
+	}
+}
```

Comment:
I don't know of a place where that function would be used.

Reply:
This was part of my broader comment series about the weird (to me) `InvalidResources` condition handling. But you would use it for the "replace existing condition" test case [above](https://github.com/openshift/hive/pull/1093/files/7d5d098fada37f7c0e976740b73de3b63aaf7487..081311b6b0065e26ec60ac110433f9e75d483849#diff-ae51ad1fa50a6b1be3a343244064484f), if you were building the syncset with chained builder funcs instead of `FullBuilder`. Feel free to ignore.